dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/test/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtilsTest.java

```diff
@@ -70,6 +72,7 @@ public void testGetPidsStr() throws Exception {
         String exceptPidsStr3 = "5000 6000 7000 7100";
         String command3;
         if (SystemUtils.IS_OS_MAC) {
+            pids3 = "-+= 5000 sshd -+- 6000 sshd --= 7000 bash --- 7100 python";
```

Comment:
May I ask what is the difference between these 3 test cases?

Reply:
These are test cases in different scenarios, such as the process ID and name may be repeated or contain some special characters.

Reply:
In fact, I think these three tests can be put together in one test case, there is no need to test three times.

Reply:
Need I put these three test cases in a single test case？

Reply:
Maybe we could test this method separately according to operating system, what do u think

Reply:
I tried to test this method separately by operating system, but found that it did not work. The operating system field in `SystemUtils` is declared as `static final`, which means we can't modify it by reflection.
```
public static final String OS_NAME = getSystemProperty("os.name");
public static final boolean IS_OS_LINUX = getOsMatchesName("Linux") || getOsMatchesName("LINUX");
public static final boolean IS_OS_MAC = getOsMatchesName("Mac");
public static final boolean IS_OS_WINDOWS = getOsMatchesName(OS_NAME_WINDOWS_PREFIX);

private static boolean getOsMatchesName(final String osNamePrefix) {
    return isOSNameMatch(OS_NAME, osNamePrefix);
}

static boolean isOSNameMatch(final String osName, final String osNamePrefix) {
    if (osName == null) {
        return false;
    }
    return osName.startsWith(osNamePrefix);
}
```